File: apps/ios/Barns/BarnsApp/Domain/Entities/CareTask.swift
import Foundation

struct CareTask: Equatable, Identifiable, Sendable {
    let id: String
    let productItemId: String
    let title: String
    let careType: CareType
    let dueDate: Date
    let repeatRule: CareRepeatRule?
    let status: CareTaskStatus
    let reminderEnabled: Bool
    let imageUrl: URL?
}

enum CareType: String, Equatable, Sendable {
    case watering
    case cleaning
    case pruning
    case inspection
    case replacement
    case other
}

enum CareRepeatRule: String, Equatable, Sendable {
    case daily
    case weekly
    case monthly
    case seasonal
}

enum CareTaskStatus: String, Equatable, Sendable {
    case pending
    case completed
    case skipped
}

File: apps/android/app/src/main/java/com/barns/app/domain/model/CareTask.kt
package com.barns.app.domain.model

import java.time.Instant

data class CareTask(
    val id: String,
    val productItemId: String,
    val title: String,
    val careType: CareType,
    val dueDate: Instant,
    val repeatRule: CareRepeatRule?,
    val status: CareTaskStatus,
    val reminderEnabled: Boolean,
    val imageUrl: String?,
)

enum class CareType { WATERING, CLEANING, PRUNING, INSPECTION, REPLACEMENT, OTHER }

enum class CareRepeatRule { DAILY, WEEKLY, MONTHLY, SEASONAL }

enum class CareTaskStatus { PENDING, COMPLETED, SKIPPED }
